app/controllers/application_controller.rb

```diff
@@ -414,6 +414,11 @@ def webpack_server
     @dev_server ||= "#{request.protocol}#{request.host}:#{port}"
   end
 
+  def compute_resource_error(listing, error)
```

Comment:
Where would you put this code? If I want to DRY this, application_controller seems right (I can move this to a concern though)